### DISCUSSION TITLE
AdGuard does not show version number

### DISCUSSION THREAD

**Discussion by chrboesch** [AdguardTeam/AdGuardHome#3046] [Q&A]
### Issue Details
AdGuard Home
Version: v0.0.0
Channel: development
Go version: go1.16.3
Build time: 2021-04-30T15:43:41Z+0000
GOOS: linux
GOARCH: arm64
Race: false
Main module:
	github.com/AdguardTeam/AdGuardHome (devel)
Dependencies:
	github.com/AdguardTeam/dnsproxy@v0.37.2 (sum: h1:3lgizD+lZI6uqxFiQykd1/hV7Ji4vSJBMejl1rbFAXU=)
	github.com/AdguardTeam/golibs@v0.4.5 (sum: h1:RRA9ZsmbJEN4OllAx0BcfvSbRBxxpWluJijBYmtp13U=)
	github.com/AdguardTeam/urlfilter@v0.14.5 (sum: h1:WyF0hg0MwKevsqNPkoaZFH8f5WRi/yuy/7qePtYt5Ts=)
	github.com/NYTimes/gziphandler@v1.1.1 (sum: h1:ZUDjpQae29j0ryrS0u/B8HZfJBtBQHjqw2rQ2cqUQ3I=)
	github.com/aead/chacha20@v0.0.0-20180709150244-8b13a72661da (sum: h1:KjTM2ks9d14ZYCvmHS9iAKVt9AyzRSqNU1qabPih5BY=)
	github.com/aead/poly1305@v0.0.0-20180717145839-3fee0db0b635 (sum: h1:52m0LGchQBBVqJRyYYufQuIbVqRawmubW3OFGqK1ekw=)
	github.com/ameshkov/dnscrypt/v2@v2.1.3 (sum: h1:DG4Uf7LSDg6XDj9sp3maxh3Ur26jeGQaP5MeYosn6v0=)
	github.com/ameshkov/dnsstamps@v1.0.3 (sum: h1:Srzik+J9mivH1alRACTbys2xOxs0lRH9qnTA7Y1OYVo=)
	github.com/beefsack/go-rate@v0.0.0-20200827232406-6cde80facd47 (sum: h1:M57m0xQqZIhx7CEJgeLSvRFKEK1RjzRuIXiA3HfYU7g=)
	github.com/cheekybits/genny@v1.0.0 (sum: h1:uGGa4nei+j20rOSeDeP5Of12XVm7TGUd4dJA9RDitfE=)
	github.com/digineo/go-ipset/v2@v2.2.1 (sum: h1:k6skY+0fMqeUjjeWO/m5OuWPSZUAn7AucHMnQ1MX77g=)
	github.com/fsnotify/fsnotify@v1.4.9 (sum: h1:hsms1Qyu0jgnwNXIxa+/V/PDsU6CfLf6CNO8H7IWoS4=)
	github.com/go-ping/ping@v0.0.0-20210216210419-25d1413fb7bb (sum: h1:2opwLSXqxE0Za64PdpskXuvLYDj/XHQAD8tLcYpSlvY=)
	github.com/gobuffalo/envy@v1.9.0 (sum: h1:eZR0DuEgVLfeIb1zIKt3bT4YovIMf9O9LXQeCZLXpqE=)
	github.com/gobuffalo/packd@v1.0.0 (sum: h1:6ERZvJHfe24rfFmA9OaoKBdC7+c9sydrytMg8SdFGBM=)
	github.com/gobuffalo/packr@v1.30.1 (sum: h1:hu1fuVR3fXEZR7rXNW3h8rqSML8EVAf6KNm0NKO/wKg=)
	github.com/google/renameio@v1.0.1-0.20210406141108-81588dbe0453 (sum: h1:vvZyWjAX9oDB+DWpMsZMyv6Q3NZtim2C5Zcdh+H0OmQ=)
	github.com/AdguardTeam/dhcp@v0.0.0-20210420175708-50b0efd52063 (sum: h1:RBsQppxEJEqHApY6WDBkM2H0UG5wt57RcT0El2WGdp8=)
	github.com/joho/godotenv@v1.3.0 (sum: h1:Zjp+RcGpHhGlrMbJzXTrZZPrWj+1vfm90La1wgB6Bhc=)
	github.com/joomcode/errorx@v1.0.3 (sum: h1:3e1mi0u7/HTPNdg6d6DYyKGBhA5l9XpsfuVE29NxnWw=)
	github.com/josharian/native@v0.0.0-20200817173448-b6b71def0850 (sum: h1:uhL5Gw7BINiiPAo24A2sxkcDI0Jt/sqp1v5xQCniEFA=)
	github.com/kardianos/service@v1.2.0 (sum: h1:bGuZ/epo3vrt8IPC7mnKQolqFeYJb7Cs8Rk4PSOBB/g=)
	github.com/lucas-clemente/quic-go@v0.20.1 (sum: h1:hb5m76V8QS/8Nw/suHvXqo3BMHAozvIkcnzpJdpanSk=)
	github.com/marten-seemann/qtls-go1-16@v0.1.3 (sum: h1:XEZ1xGorVy9u+lJq+WXNE+hiqRYLNvJGYmwfwKQN2gU=)
	github.com/mdlayher/ethernet@v0.0.0-20190606142754-0394541c37b7 (sum: h1:lez6TS6aAau+8wXUP3G9I3TGlmPFEq2CTxBaRqY6AGE=)
	github.com/mdlayher/netlink@v1.4.0 (sum: h1:n3ARR+Fm0dDv37dj5wSWZXDKcy+U0zwcXS3zKMnSiT0=)
	github.com/mdlayher/raw@v0.0.0-20191009151244-50f2db8cc065 (sum: h1:aFkJ6lx4FPip+S+Uw4aTegFMct9shDvP+79PsSxpm3w=)
	github.com/miekg/dns@v1.1.40 (sum: h1:pyyPFfGMnciYUk/mXpKkVmeMQjfXqt3FAJ2hy7tPiLA=)
	github.com/patrickmn/go-cache@v2.1.0+incompatible (sum: h1:HRMgzkcYKYpi3C8ajMPV8OFXaaRUnok+kx1WdO15EQc=)
	github.com/pkg/errors@v0.9.1 (sum: h1:FEBLx1zS214owpjy7qsBeixbURkuhQAwrK5UwLGTwt4=)
	github.com/rogpeppe/go-internal@v1.7.0 (sum: h1:3qqXGV8nn7GJT65debw77Dzrx9sfWYgP0DDo7xcMFRk=)
	github.com/satori/go.uuid@v1.2.0 (sum: h1:0uYX9dsZ2yD7q2RtLRtPSdGDWzjeM3TbMJP9utgA0ww=)
	github.com/ti-mo/netfilter@v0.4.0 (sum: h1:rTN1nBYULDmMfDeBHZpKuNKX/bWEXQUhe02a/10orzg=)
	github.com/u-root/u-root@v7.0.0+incompatible (sum: h1:u+KSS04pSxJGI5E7WE4Bs9+Zd75QjFv+REkjy/aoAc8=)
	go.etcd.io/bbolt@v1.3.5 (sum: h1:XAzx9gjCb0Rxj7EoqcClPD1d5ZBxZJk0jbuoPHenBt0=)
	golang.org/x/crypto@v0.0.0-20210220033148-5ea612d1eb83 (sum: h1:/ZScEX8SfEmUGRHs0gxpqteO5nfNW6axyZbBdw9A12g=)
	golang.org/x/net@v0.0.0-20210405180319-a5a99cb37ef4 (sum: h1:4nGaVu0QrbjT/AK2PRLuQfQuh6DJve+pELhqTdAj3x0=)
	golang.org/x/sys@v0.0.0-20210330210617-4fbd30eecc44 (sum: h1:Bli41pIlzTzf3KEY06n+xnzK/BESIg2ze4Pgfh/aI8c=)
	golang.org/x/text@v0.3.5 (sum: h1:i6eZZ+zk0SOf0xgBpEpPD18qWcJda6q1sxt3S0kzyUQ=)
	gopkg.in/natefinch/lumberjack.v2@v2.0.0 (sum: h1:1Lc07Kr7qY4U2YPouBjpCLxpiyxIVoxqXgkXLknAOE8=)
	gopkg.in/yaml.v2@v2.4.0 (sum: h1:D8xgwECY7CYvx+Y2n4sBz93Jn9JRvxdiyyo8CTfuKaY=)
	howett.net/plist@v0.0.0-20201203080718-1454fab16a06 (sum: h1:QDxUo/w2COstK1wIBYpzQlHX/NqaQTcf9jyz347nI58=)

### Expected Behavior
"AdGuardHome -v" should show version number v0.106.1

### Actual Behavior
"AdGuardHome -v" shows v0.0.0

**Comment by ainar-g** [comment]
We cannot reproduce this with the released binaries.  Is this a custom-built one?

**Comment by chrboesch** [comment]
Yes, compiled directly on the Raspi with OS Arch Linux 64bit.

**Comment by chrboesch** [comment]
==> Making package: adguardhome 1:0.106.1-1 (Fri Apr 30 15:32:04 2021)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Downloading adguardhome-0.106.1.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   133  100   133    0     0    465      0 --:--:-- --:--:-- --:--:--   466
100 1833k    0 1833k    0     0  1616k      0 --:--:--  0:00:01 --:--:-- 5283k
  -> Found AdGuardHome.service
  -> Found sysusers.conf
  -> Found tmpfiles.conf
==> Validating source files with b2sums...
    adguardhome-0.106.1.tar.gz ... Passed
    AdGuardHome.service ... Passed
    sysusers.conf ... Passed
    tmpfiles.conf ... Passed
==> Extracting sources...
  -> Extracting adguardhome-0.106.1.tar.gz with bsdtar
==> Removing existing $pkgdir/ directory...
==> Starting build()...
npm --prefix client --quiet --no-progress --ignore-engines --ignore-optional --ignore-platform --ignore-scripts ci

added 1828 packages, and audited 1829 packages in 55s

16 vulnerabilities (10 low, 2 moderate, 4 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
yarn --cwd client2 --network-timeout 120000 --silent --ignore-engines --ignore-optional --ignore-platform --ignore-scripts install
warning "recharts > react-resize-detector@5.2.0" has incorrect peer dependency "react@^16.0.0".
warning "recharts > react-resize-detector@5.2.0" has incorrect peer dependency "react-dom@^16.0.0".
warning "recharts > react-smooth@1.0.6" has incorrect peer dependency "react@^15.0.0 || ^16.0.0".
warning "recharts > react-smooth@1.0.6" has incorrect peer dependency "react-dom@^15.0.0 || ^16.0.0".
warning "eslint-config-airbnb-typescript > eslint-config-airbnb@18.2.0" has unmet peer dependency "eslint-plugin-jsx-a11y@^6.3.0".
warning " > less-loader@5.0.0" has incorrect peer dependency "webpack@^2.0.0 || ^3.0.0 || ^4.0.0".
warning " > optimize-css-assets-webpack-plugin@5.0.4" has incorrect peer dependency "webpack@^4.0.0".
warning Ignored scripts due to flag.
env COMMIT='$( git rev-parse --short HEAD )' CHANNEL='development' GPG_KEY='devteam@adguard.com' GPG_KEY_PASSPHRASE='not-a-real-password' DIST_DIR='dist' GO="${GO:-go}" GOPROXY='https://goproxy.cn|https://proxy.golang.org|direct' PATH="${P
WD}/bin:$( "${GO:-go}" env GOPATH )/bin:${PATH}" RACE='0' SIGN='1' VERBOSE='0' VERSION='v0.0.0'  "/bin/sh" ./scripts/make/go-deps.sh
npm --prefix client run build-prod

> dashboard@0.1.0 build-prod
> cross-env BUILD_ENV=prod webpack --config webpack.prod.js

   1134 modules

WARNING in 
src/components/Logs/Logs.css
 174:5   ⚠  Expected empty line before declaration   declaration-empty-line-before
 439:18  ⚠  Expected numeric font-weight notation    font-weight-notation


yarn --cwd client2 build
yarn run v1.22.10
$ webpack --config ./scripts/webpack/webpack.config.prod.js
(node:8890) [DEP_WEBPACK_COMPILATION_OPTIMIZE_CHUNK_ASSETS] DeprecationWarning: optimizeChunkAssets is deprecated (use Compilation.hook.processAssets instead and use one of Compilation.PROCESS_ASSETS_STAGE_* as stage option)
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:8890) [DEP0148] DeprecationWarning: Use of deprecated folder mapping "./" in the "exports" field module resolution of the package at /home/chris/aur/adguardhome/src/AdGuardHome-0.106.1/client2/node_modules/postcss/package.json.
Update this package.json to use a subpath pattern like "./*".
(node:8890) [DEP_WEBPACK_TEMPLATE_PATH_PLUGIN_REPLACE_PATH_VARIABLES_HASH] DeprecationWarning: [hash] is now [fullhash] (also consider using [chunkhash] or [contenthash], see documentation for details)
(node:8890) [DEP_WEBPACK_COMPILATION_ASSETS] DeprecationWarning: Compilation.assets will be frozen in future, all modifications are deprecated.
BREAKING CHANGE: No more changes should happen to Compilation.assets after sealing the Compilation.
        Do changes to assets earlier, e. g. in Compilation.hooks.processAssets.
        Make sure to select an appropriate stage from Compilation.PROCESS_ASSETS_STAGE_*.
7 assets
3320 modules
webpack 5.18.0 compiled successfully in 278351 ms
Done in 288.01s.
env COMMIT='$( git rev-parse --short HEAD )' CHANNEL='development' GPG_KEY='devteam@adguard.com' GPG_KEY_PASSPHRASE='not-a-real-password' DIST_DIR='dist' GO="${GO:-go}" GOPROXY='https://goproxy.cn|https://proxy.golang.org|direct' PATH="${P
WD}/bin:$( "${GO:-go}" env GOPATH )/bin:${PATH}" RACE='0' SIGN='1' VERBOSE='0' VERSION='v0.0.0'  "/bin/sh" ./scripts/make/go-build.sh
==> Entering fakeroot environment...
==> Starting package()...
==> Tidying install...
  -> Removing libtool files...
  -> Purging unwanted files...
  -> Removing static library files...
  -> Stripping unneeded symbols from binaries and libraries...
  -> Compressing man and info pages...
==> Checking for packaging issues...
==> WARNING: Package contains reference to $srcdir
var/lib/adguardhome/AdGuardHome
==> Creating package "adguardhome"...
  -> Generating .PKGINFO file...
  -> Generating .BUILDINFO file...
  -> Adding install file...
  -> Generating .MTREE file...
  -> Compressing package...
==> Leaving fakeroot environment.
==> Finished making: adguardhome 1:0.106.1-1 (Fri Apr 30 15:44:52 2021)
==> Installing package adguardhome with pacman -U...

**Reply by ainar-g** [reply]
If you want to build your own AdGuard Home and make it as close as possible to the released binaries, please refer to our build script documentation [here].  In your case, it's probably the fact that you haven't set the `CHANNEL` macro to `release` when invoking `make`.

[here]: https://github.com/AdguardTeam/AdGuardHome/tree/master/scripts#make-makefile-scripts

**Reply by chrboesch** [reply]
Thank you very much, I will forward this info to the package maintainer.
